src/NodeApi/Runtime/NodejsPlatform.cs

```diff
@@ -25,23 +26,34 @@ public sealed class NodejsPlatform : IDisposable
     /// <summary>
     /// Initializes the Node.js platform.
     /// </summary>
-    /// <param name="libnodePath">Path to the `libnode` shared library, including extension.</param>
+    /// <param name="libnode">
+    /// Name of the `libnode` shared library.
+    /// Has to be a full file path when using .NET Framework.
+    /// </param>
     /// <param name="args">Optional platform arguments.</param>
     /// <exception cref="InvalidOperationException">A Node.js platform instance has already been
     /// loaded in the current process.</exception>
     public NodejsPlatform(
-        string libnodePath,
+        string libnode,
         string[]? args = null)
     {
-        if (string.IsNullOrEmpty(libnodePath)) throw new ArgumentNullException(nameof(libnodePath));
-
         if (Current != null)
         {
             throw new InvalidOperationException(
                 "Only one Node.js platform instance per process is allowed.");
         }
 
-        nint libnodeHandle = NativeLibrary.Load(libnodePath);
+        var entryAssembly = Assembly.GetAssembly(typeof(NodejsPlatform));
+
+        nint libnodeHandle =
+            entryAssembly != null
+                ? NativeLibrary.Load(
+                    libnode,
+                    entryAssembly!,
+                    DllImportSearchPath.AssemblyDirectory | DllImportSearchPath.SafeDirectories
```

Comment:
I can do that, but after some research it looks like CA5393 might have some problems in it: It looks like "SafeDirectories" includes the application directory, but then "ApplicationDirectory" is considered unsafe. Maybe this rule hasn't been updated in a while, or maybe it was made with only Windows in mind? On UNIX it's quite common to load libraries from the assembly directory and if it really was insecure, wouldn't that mean that .NETs own approach of putting managed dependency DLLs next to the assembly is also insecure?

If you want I can still change this but to me, the rule looks a little weird.

Reply:
I learned today that the `DllImportSearchPath` is entirely ignored on non-Windows platforms, and that is not documented: https://github.com/dotnet/dotnet-api-docs/issues/3205

In that case should we use your `FindLibrary()` code on those platforms? Or if not, will it even search in the right place(s)? It probably requires testing.

Also I found that CA5393 is now [disabled by default](https://github.com/dotnet/roslyn-analyzers/issues/2855) since it doesn't apply cross-platform, so maybe we don't need to worry about that. (I recalled dealing with this warning in other projects several years ago.)

Reply:
At least for the near future, loading `libnode` from a system directory is rarely going to be the right choice. The Node.js project does not publish official libnode shared-library binaries, and I believe most applications that do use it now are static-linked. Also, applications are likely to be sensitive to the major version of node. So any apps that do use a libnode shared-library will distribute it with the application.

For this project, our plan is to publish libnode binaries via NuGet packages. Then when a .NET application that consumes the package is published, the libnode binary for the target runtime-identifier would get copied to the publish output directory. That would be the assembly directory, or at least relative to it. So I'm thinking the best default should probably be just `AssemblyDirectory`, and we can let the application override that with an optional parameter as I suggested above.

We are working on setting up that infrastructure for building libnode, packaging multiple platforms in nuget packages, and publishing them. When that is further along, it might be necessary to make adjustments to the loading code here anyway.